aws/internal/service/xray/waiter/waiter.go

```diff
@@ -0,0 +1,26 @@
+package waiter
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// EncryptionConfigAvailable waits for a EncryptionConfig to return Available
+func EncryptionConfigAvailable(conn *xray.XRay) (*xray.EncryptionConfig, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{xray.EncryptionStatusUpdating},
+		Target:  []string{xray.EncryptionStatusActive},
+		Refresh: EncryptionConfigStatus(conn),
+		Timeout: 15 * time.Minute,
```

Comment:
Nit: Timeout should be moved to new constant, e.g. `EncryptionConfigAvailableTimeout = 15 * time.Minute`